lib/modules/manager/kustomize/artifacts.spec.ts

```diff
@@ -0,0 +1,691 @@
+        packageFileName,
+        updatedDeps,
+        newPackageFileContent,
+        config,
+      }),
+    ).toBeNull();
+  });
+
+  it('returns null if no packageName is found', async () => {
+    const updatedDeps = [
+      {
+        depType: 'HelmChart',
+        depName: 'example',
+        newVersion: '2.0.0',
+        currentVersion: '1.0.0',
+        packageName: undefined,
+        datasource: DockerDatasource.id,
+      },
+    ];
+    expect(
+      await kustomize.updateArtifacts({
+        packageFileName,
+        updatedDeps,
+        newPackageFileContent,
+        config,
+      }),
+    ).toBeNull();
+  });
+
+  it('returns null if neither currentVersion or newVersion is found', async () => {
+    const updatedDeps = [
+      {
+        depType: 'HelmChart',
+        depName: 'example',
+        newVersion: undefined,
+        currentVersion: undefined,
+        registryUrls: ['https://github.com.com/example/example'],
+        datasource: HelmDatasource.id,
+      },
+    ];
+    expect(
+      await kustomize.updateArtifacts({
+        packageFileName,
+        updatedDeps,
+        newPackageFileContent,
+        config,
+      }),
+    ).toBeNull();
+  });
+
+  it('returns null if newVersion is not found and currentVersion is already inflated', async () => {
+    fs.localPathExists.mockResolvedValueOnce(true);
+    fs.localPathExists.mockResolvedValueOnce(true);
+    const updatedDeps = [
+      {
+        depType: 'HelmChart',
+        depName: 'example',
+        newVersion: undefined,
+        currentVersion: '1.0.0',
+        registryUrls: ['https://github.com.com/example/example'],
+        datasource: HelmDatasource.id,
+      },
+    ];
+    expect(
+      await kustomize.updateArtifacts({
+        packageFileName,
+        updatedDeps,
+        newPackageFileContent,
+        config,
+      }),
+    ).toBeNull();
+  });
+
+  it('returns null if old version is not inflated and kustomizeInflateHelmCharts is not enabled', async () => {
+    const execSnapshots = mockExecAll();
+
+    fs.localPathExists.mockResolvedValueOnce(false);
+    git.getRepoStatus.mockResolvedValueOnce(
+      partial<StatusResult>({
+        not_added: [],
+        deleted: [],
+      }),
+    );
+    const updatedDeps = [
+      {
+        depType: 'HelmChart',
+        depName: 'example',
+        newVersion: '2.0.0',
+        currentVersion: '1.0.0',
+        registryUrls: ['https://github.com.com/example/example'],
+        datasource: HelmDatasource.id,
+      },
+    ];
+    expect(
+      await kustomize.updateArtifacts({
+        packageFileName,
+        updatedDeps,
+        newPackageFileContent,
+        config: {
+          ...config,
+          postUpdateOptions: [],
+        },
+      }),
+    ).toBeNull();
+    expect(execSnapshots).toBeEmptyArray();
+  });
+
+  it('returns null if newVersion and currentVersion is the same', async () => {
+    fs.readLocalFile.mockResolvedValueOnce(null);
```

Comment:
still needed? Default value is `undefined` which should also do it. So no need to mock a value here and below
```suggestion
```